php-mode.el

```diff
@@ -294,9 +294,12 @@ You can replace \"en\" with your ISO language code."
   :type 'string)
 
 ;;;###autoload
-(add-to-list 'interpreter-mode-alist
-             ;; Match php, php-3, php5, php7, php5.5, php-7.0.1, etc.
-             (cons "php\\(?:-?[3457]\\(?:\\.[0-9]+\\)*\\)?" 'php-mode))
+(if (version< emacs-version "24.4")
+    (dolist (i '("php" "php3" "php5" "php7" "php-5" "php-5.5" "php7.0.1"))
```

Comment:
I think that it would be better to leave only `php`, `php5` and `php7`.

Other elements are redundant as they contain minor numbers.

Reply:
The test case (`php-mode-test-issue-357`) checks that all of them work, so I didn't want to change that.

Reply:
I understand.  But, it is unreasonable to add "php 7.0.1" to the user's runtime only for testing.

We can adapt the test to the actual situation.  For example...

```patch
diff --git a/php-mode-test.el b/php-mode-test.el
index 7f498ae..7bf1ee2 100644
--- a/php-mode-test.el
+++ b/php-mode-test.el
@@ -952,11 +952,14 @@ style from Drupal."

 (ert-deftest php-mode-test-issue-357 ()
   "Match version-specific interpreters."
-  (dolist (on '("php" "php3" "php5" "php7" "php-5" "php-5.5" "php7.0.1"))
-    (with-temp-buffer
-      (insert "#!" on)
-      (set-auto-mode)
-      (should (eq 'php-mode major-mode))))
+  (let ((versions (if (version< emacs-version "24.4")
+                      '("php" "php5" "php7")
+                    '("php" "php3" "php5" "php7" "php-5" "php-5.5" "php7.0.1"))))
+    (dolist (on versions)
+      (with-temp-buffer
+        (insert "#!" on)
+        (set-auto-mode)
+        (should (eq 'php-mode major-mode)))))

   (dolist (off '("php2" "xphp5" "foo" "php8" "php7x"))
     (with-temp-buffer
```

Reply:
Well, the tests are supposed to cover actual use cases, the split would introduce different behaviour in different Emacs versions, and after all the performance implications of three vs. seven additional entries in `interpreter-mode-alist` are not measurable :-).  I'm not totally opposed to your suggestion, but I find "unreasonable" a pretty harsh judgment for adding four (!) maybe superfluous strings to a list that is not used very often.

Reply:
Hmm...  `"php7.0.1"` is one of the cases the pattern dynamically accepts, but adding it with a literal has a different meaning.  In most environments `php7.0.1` does not exist and it does not help anybody.

Reply:
We know [many officially released PHP versions](http://php.net/downloads.php), but adding them all to this list is nonsense.